IMPORTS

In [82]:
import main
import importlib
import scrape.harris.harris_county_scraper
from dotenv import load_dotenv
import ocr.ocr
import os
import string

RELOAD MODULES

In [150]:
load_dotenv()
importlib.reload(main)
importlib.reload(ocr.ocr)
importlib.reload(scrape.harris.harris_county_scraper)

<module 'scrape.harris.harris_county_scraper' from '/Users/isaaclam/guardian/marketing_leads_project/main/scrape/harris/harris_county_scraper.py'>

RUN MAIN

In [73]:
# run main
main.main()

Processing case number: 1243759...
Document matching criteria not found or not downloaded.
adding case_number: 1243759
Navigating back to search results...
Processing case number: 1243623...
Document matching criteria not found or not downloaded.
adding case_number: 1243623
Navigating back to search results...
Processing case number: 1243614...
[2025-02-04 13:30:39] Downloading - Lead_Petition_eABC_filing_package.pdf to /Users/isaaclam/guardian/marketing_leads_project/main/out/harris/downloaded_docs
Combined details: "Kendria Micheaux", "21327 Sugarbridge Ln Tomball TX 77375", "Credit Corp Solutions Inc.", "HUFFMAN, ALESHIA RENEE SCOTT & ASSOCIATES, P.C., P.O. Box 115220 Carrolton TX 75011 Fax 214-234-8554"
Extracting text from PDF...
Processing page 1...
Processing page 2...
Processing page 3...
Processing page 4...
Processing page 5...
Processing page 6...
Processing page 7...
Processing page 8...
Processing page 9...
Processing page 10...
Processing page 11...
Processing page 12...


ElementClickInterceptedException: Message: element click intercepted: Element <a class="pgr" href="javascript:__doPostBack('ctl00$ContentPlaceHolder1$DataPagerLisViewCases1$ctl03$ctl00','')">...</a> is not clickable at point (143, 8). Other element would receive the click: <div class="navbar-brand hidden-md">...</div>
  (Session info: chrome=132.0.6834.160)
Stacktrace:
0   chromedriver                        0x000000010304b0d4 cxxbridge1$str$ptr + 2600792
1   chromedriver                        0x00000001030439f0 cxxbridge1$str$ptr + 2570356
2   chromedriver                        0x0000000102be43d8 cxxbridge1$string$len + 89376
3   chromedriver                        0x0000000102c2ed44 cxxbridge1$string$len + 394892
4   chromedriver                        0x0000000102c2d320 cxxbridge1$string$len + 388200
5   chromedriver                        0x0000000102c2b1fc cxxbridge1$string$len + 379716
6   chromedriver                        0x0000000102c2a62c cxxbridge1$string$len + 376692
7   chromedriver                        0x0000000102c1f200 cxxbridge1$string$len + 330568
8   chromedriver                        0x0000000102c1eca4 cxxbridge1$string$len + 329196
9   chromedriver                        0x0000000102c63638 cxxbridge1$string$len + 610176
10  chromedriver                        0x0000000102c1d34c cxxbridge1$string$len + 322708
11  chromedriver                        0x0000000102c1df94 cxxbridge1$string$len + 325852
12  chromedriver                        0x0000000103016238 cxxbridge1$str$ptr + 2384060
13  chromedriver                        0x0000000103019550 cxxbridge1$str$ptr + 2397140
14  chromedriver                        0x0000000102ffcb9c cxxbridge1$str$ptr + 2279968
15  chromedriver                        0x0000000103019e10 cxxbridge1$str$ptr + 2399380
16  chromedriver                        0x0000000102fee6a4 cxxbridge1$str$ptr + 2221352
17  chromedriver                        0x00000001030347d8 cxxbridge1$str$ptr + 2508380
18  chromedriver                        0x0000000103034954 cxxbridge1$str$ptr + 2508760
19  chromedriver                        0x0000000103043664 cxxbridge1$str$ptr + 2569448
20  libsystem_pthread.dylib             0x000000018d8332e4 _pthread_start + 136
21  libsystem_pthread.dylib             0x000000018d82e0fc thread_start + 8


TEST PDF PROCESSING

In [74]:
ocr.ocr.process_pdf_and_find_damages('ocr/example_docs/sample1.pdf')

FileNotFoundError: PDF file not found. Please check the path.

TEST CSV CONVERSION

In [75]:
input_txt = '/Users/isaaclam/guardian/marketing_leads_project/main/out/harris/defendant_data.txt'
output_csv = '/Users/isaaclam/guardian/marketing_leads_project/main/out/harris/defendant_data.csv'
main.convert_txt_to_csv(input_txt, output_csv)

TEST PRE-PROCESSING

In [118]:
pdf_path = '/Users/isaaclam/guardian/marketing_leads_project/main/ocr/example_docs/sample1.pdf'
output_folder = 'processed_images'
pages = ocr.ocr.convert_from_path(pdf_path, dpi=300)
for page_number,page_image in enumerate(pages,start=1):
    ocr.ocr.preprocess_image_to_remove_watermark(page_image, output_folder, page_number)

In [150]:
pdf_path = '/Users/isaaclam/guardian/marketing_leads_project/main/ocr/example_docs/sample1.pdf'

if not os.path.exists(pdf_path):
    raise FileNotFoundError("PDF file not found. Please check the path.")

print("Extracting text from PDF...")
extracted_text = ocr.ocr.extract_text_from_pdf_with_watermark_removal(pdf_path)

print("\nSearching for 'damages' and the associated dollar value...")
# result = find_damages_and_value(extracted_text)

Extracting text from PDF...
Processing page 1...
Processing page 2...
Processing page 3...
Processing page 4...
Processing page 5...
Processing page 6...
Processing page 7...
Processing page 8...
Processing page 9...
Processing page 10...
Processing page 11...
Processing page 12...
Processing page 13...
Processing page 14...
Processing page 15...
Processing page 16...
Processing page 17...
Processing page 18...
Processing page 19...
Processing page 20...
Processing page 21...
Processing page 22...
Processing page 23...
Processing page 24...
Processing page 25...
Processing page 26...
Processing page 27...
Processing page 28...
Processing page 29...
Processing page 30...
Processing page 31...
Processing page 32...
Processing page 33...
Processing page 34...
Processing page 35...
Processing page 36...
Processing page 37...
Processing page 38...
Processing page 39...
Processing page 40...
Processing page 41...
Processing page 42...
Processing page 43...
Processing page 44...
Processing pa

MANUALLY VERIFY CSV

In [80]:
verified_csv = '/Users/isaaclam/guardian/marketing_leads_project/main/out/harris/defendant_data_verified.csv'
output_csv = '/Users/isaaclam/guardian/marketing_leads_project/main/out/harris/defendant_data.csv'
filtered_csv = '/Users/isaaclam/guardian/marketing_leads_project/main/out/harris/defendant_data_verified_filtered.csv'
main.verify_csv(output_csv, verified_csv,filtered_csv)

TEST OCR AGAINST MANUALLY CREATED DATASET

In [65]:
def getDollarAmount(input):
    dollar_amount = None
    for word in input.split():
        if word.startswith('$'):
            # Strip trailing punctuation (but keep the initial '$')
            dollar_amount = word.rstrip(string.punctuation)
            break
    return dollar_amount if dollar_amount else "No dollar amount found."

In [58]:
mainDir = '/Users/isaaclam/guardian/marketing_leads_project/main'
outDir = f'{mainDir}/test/ocr/out'
pdfDir = f'{mainDir}/test/ocr/test_data_set'

In [155]:
with open(f'{mainDir}/test/ocr/test_results_dataset.txt','r',encoding='utf-8') as f:
    lines = f.readlines()

In [ ]:
for i in range (1,90):
    pdfStr =f'{pdfDir}/{i}.pdf'
    res = ocr.ocr.process_pdf_and_find_damages(pdfStr,False)
    
    dollarAmount = getDollarAmount(res)
    splitRes = res.split(',')
    courtName = splitRes[len(splitRes)-1]
    courtNameSplit = courtName.split(' ')
    courtNumber = courtNameSplit[len(courtNameSplit)-2]
    with open(f'{outDir}/out.txt', 'a', encoding = 'utf-8') as f:
        appendStr = f'{dollarAmount}, {courtNumber}\n'
        eq = appendStr == lines[i-1]
        f.write(f'{appendStr[:len(appendStr)-1]}, {eq}\n')


    print(f'{dollarAmount}, {courtName}, {courtNumber}')



Extracting text from PDF...
Processing page 1...
Processing page 2...
Processing page 3...
Processing page 4...
Processing page 5...

Searching for 'damages' and the associated dollar value...
appendstr: $146,600.00, 1
 lines[i-1]: $146,600.00, 1
 eq: True
$146,600.00,  "Harris County - County Civil Court at Law No. 1 ", 1
Extracting text from PDF...
Processing page 1...
Processing page 2...
Processing page 3...
Processing page 4...
Processing page 5...

Searching for 'damages' and the associated dollar value...
appendstr: $20,134.58, 3
 lines[i-1]: $20,134.58, 3
 eq: True
$20,134.58,  "Harris County - County Civil Court at Law No. 3 ", 3
Extracting text from PDF...
Processing page 1...
Processing page 2...
Processing page 3...
Processing page 4...
Processing page 5...

Searching for 'damages' and the associated dollar value...
appendstr: $23,053.38, 3
 lines[i-1]: $23,053.38, 3
 eq: True
$23,053.38,  "Harris County - County Civil Court at Law No. 3 ", 3
Extracting text from PDF...
Pro

KeyboardInterrupt: 

LIST INCORRECT ANSWERS, AND HOW MANY CORRECT ANSWERS

In [91]:
with open(f'{mainDir}/test/ocr/out/out.txt','r',encoding='utf-8') as f:
    lines = f.readlines()
    numPos = 0
    numLines = len(lines)
    for i in range(len(lines)):
        splitString = lines[i].split(', ')
        n = len(splitString)
        if splitString[n-1] == 'True\n':
            numPos += 1
        else:
            print(f'INCORRECT: {i} {lines[i]}' )
    print(f'{numPos}/{numLines} ({numPos/numLines}) correct')

INCORRECT: 18 $250,000.00, 2, False

INCORRECT: 19 $250,000.00, 4, False

INCORRECT: 21 $250,000.00, 3, False

INCORRECT: 22 $250,000.00, 1, False

INCORRECT: 24 $250,000.00, 1, False

INCORRECT: 33 No dollar amount found., 4, False

INCORRECT: 36 No dollar amount found., 4, False

INCORRECT: 56 $4,396.00, 1243129, False

INCORRECT: 61 $610.51, 1, False

INCORRECT: 67 $3,668.29, 1, False

INCORRECT: 76 $2,613.54, 1, False

INCORRECT: 86 $250,000.00, 4, False

INCORRECT: 87 $250,000.00, 2, False

76/89 (0.8539325842696629) correct


examples that would not be flagged:  
  
62.pdf INCORRECT: $610.51, 1, False

68.pdf INCORRECT: $3,668.29, 1, False

77.pdf INCORRECT: $2,613.54, 1, False

In [147]:
for i in [62,68,77]:
    pdfStr =f'{pdfDir}/{i}.pdf'
    res = ocr.ocr.process_pdf_and_find_damages(pdfStr,False)
    
    dollarAmount = getDollarAmount(res)
    splitRes = res.split(',')
    courtName = splitRes[len(splitRes)-1]
    courtNameSplit = courtName.split(' ')
    courtNumber = courtNameSplit[len(courtNameSplit)-2]
    with open(f'{outDir}/out.txt', 'a', encoding = 'utf-8') as f:
        appendStr = f'{dollarAmount}, {courtNumber}\n'
        eq = appendStr == lines[i-1]
        f.write(f'{appendStr[:len(appendStr)-1]}, {eq}\n')


    print(f'{dollarAmount}, {courtName}, {courtNumber}')

Extracting text from PDF...
Processing page 1...
Processing page 2...
Processing page 3...
Processing page 4...

Searching for 'damages' and the associated dollar value...
$610.51,  "Harris County - County Civil Court at Law No. -1 ", -1
Extracting text from PDF...
Processing page 1...
Processing page 2...
Processing page 3...
Processing page 4...

Searching for 'damages' and the associated dollar value...
$3,668.29,  "Harris County - County Civil Court at Law No. -1 ", -1
Extracting text from PDF...
Processing page 1...
Processing page 2...
Processing page 3...
Processing page 4...

Searching for 'damages' and the associated dollar value...
$2,613.54,  "Harris County - County Civil Court at Law No. -1 ", -1
